File: 14-longest-common-prefix/longest-common-prefix.java
class Solution {
    public String longestCommonPrefix(String[] strs) {
        String prefix="";
        for(int i=0; i < strs[0].length(); i++)
        {
            char c= strs[0].charAt(i);
             for(int j=1; j<strs.length; j++)
            {
                if(i > strs[j].length()-1 || strs[j].charAt(i) != c)
                return prefix;
            }
            prefix= prefix+ c;
        }
        return prefix;
    }
}

File: 14-longest-common-prefix/longest-common-prefix.py
class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        prefix=''
        k=0
        for i in strs[0]:
            flag=True
            for j in range(1, len(strs)):
                if k>= len(strs[j]) or strs[j][k]!=i:
                    flag=False
                    break
            if flag==True:
                prefix+=(i)
                k+=1
            else:
                break                                
        return prefix